File: src/main/java/com/josephbaca/rpggame/ContextManager.java
package com.josephbaca.rpggame;


import com.josephbaca.world.Context;

import java.util.ArrayList;
import java.util.List;

/**
 * Internal representation of the context to manage current game context.
 */
public class ContextManager {

  private final List<Context> contextStack = new ArrayList<>();

  public ContextManager() { }

  public void addContextLayer(Context c) {
    contextStack.add(c);
  }

  public void removeContextLayer() {
    contextStack.remove(contextStack.size() - 1);
  }

  public void replaceContextLayer(Context c) {
    removeContextLayer();
    addContextLayer(c);
  }

  public Context getCurrentContext() {
    return contextStack.get(contextStack.size() - 1);
  }
}

File: src/main/java/com/josephbaca/rpggame/ContextManager.kt
package com.josephbaca.rpggame


import com.josephbaca.world.Context

import java.util.ArrayList

/**
 * Internal representation of the contextManager to manage current game contextManager.
 */
class ContextManager {

    val contextStack = ArrayList<Context>()

    val currentContext: Context
        get() = contextStack.last()

    fun addContextLayer(c: Context) {
        contextStack.add(c)
    }

    fun removeContextLayer() {
        contextStack.removeAt(contextStack.size - 1)
    }

    fun replaceContextLayer(c: Context) {
        removeContextLayer()
        addContextLayer(c)
    }

    fun insertContextLayer(c: Context, index: Int) {
        contextStack.add(index, c)
    }
}
